core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestSimplifyExpressions.java

```diff
@@ -121,158 +163,163 @@ public void testExtractCommonPredicates()
+        assertSimplifies("CAST(REAL '0e0' / REAL '0e0' AS varchar(3))", "'NaN'", createVarcharType(3));
+        assertSimplifies("CAST(REAL 'Infinity' AS varchar(8))", "'Infinity'", createVarcharType(8));
+        assertSimplifies("CAST(REAL '12e2' AS varchar(5))", "'1.2E3'", createVarcharType(5));
+        assertSimplifies("CAST(REAL '-12e2' AS varchar(50))", "CAST('-1.2E3' AS varchar(50))", createVarcharType(50));
 
         // cast from real to varchar fails, so the expression is not modified
-        assertSimplifies("CAST(REAL '12e2' AS varchar(3))", "CAST(REAL '12e2' AS varchar(3))");
-        assertSimplifies("CAST(REAL '-12e2' AS varchar(3))", "CAST(REAL '-12e2' AS varchar(3))");
-        assertSimplifies("CAST(REAL 'NaN' AS varchar(2))", "CAST(REAL 'NaN' AS varchar(2))");
-        assertSimplifies("CAST(REAL 'Infinity' AS varchar(7))", "CAST(REAL 'Infinity' AS varchar(7))");
-        assertSimplifies("CAST(REAL '12e2' AS varchar(3)) = '1200.0'", "CAST(REAL '12e2' AS varchar(3)) = '1200.0'");
+        assertSimplifies("CAST(REAL '12e2' AS varchar(3))", "CAST(REAL '12e2' AS varchar(3))", createVarcharType(3));
+        assertSimplifies("CAST(REAL '-12e2' AS varchar(3))", "CAST(REAL '-12e2' AS varchar(3))", createVarcharType(3));
+        assertSimplifies("CAST(REAL 'NaN' AS varchar(2))", "CAST(REAL 'NaN' AS varchar(2))", createVarcharType(2));
+        assertSimplifies("CAST(REAL 'Infinity' AS varchar(7))", "CAST(REAL 'Infinity' AS varchar(7))", createVarcharType(7));
+        assertSimplifies("CAST(REAL '12e2' AS varchar(3)) = '1200.0'", "CAST(REAL '12e2' AS varchar(3)) = '1200.0'", createVarcharType(3));
     }
 
     @Test
     public void testCastDateToBoundedVarchar()
     {
         // the varchar type length is enough to contain the date's representation
-        assertSimplifies("CAST(DATE '2013-02-02' AS varchar(10))", "'2013-02-02'");
-        assertSimplifies("CAST(DATE '2013-02-02' AS varchar(50))", "CAST('2013-02-02' AS varchar(50))");
+        assertSimplifies("CAST(DATE '2013-02-02' AS varchar(10))", "'2013-02-02'", createVarcharType(10));
+        assertSimplifies("CAST(DATE '2013-02-02' AS varchar(50))", "CAST('2013-02-02' AS varchar(50))", createVarcharType(50));
 
         // cast from date to varchar fails, so the expression is not modified
-        assertSimplifies("CAST(DATE '2013-02-02' AS varchar(3))", "CAST(DATE '2013-02-02' AS varchar(3))");
-        assertSimplifies("CAST(DATE '2013-02-02' AS varchar(3)) = '2013-02-02'", "CAST(DATE '2013-02-02' AS varchar(3)) = '2013-02-02'");
+        assertSimplifies("CAST(DATE '2013-02-02' AS varchar(3))", "CAST(DATE '2013-02-02' AS varchar(3))", createVarcharType(3));
+        assertSimplifies("CAST(DATE '2013-02-02' AS varchar(3)) = '2013-02-02'", "CAST(DATE '2013-02-02' AS varchar(3)) = '2013-02-02'", createVarcharType(3));
+    }
+
+    private static void assertSimplifiesWithBooleanType(@Language("SQL") String expression, @Language("SQL") String expected)
```

Comment:
We don't need a special method for boolean. Inline this method and pass the type explicitly, just like every other call to `assertSimplifies`